tiatoolbox/dataloader/wsireader.py

```diff
@@ -0,0 +1,101 @@
+"""
+WSIReader for WSI reading or extracting metadata information from WSIs
+"""
+import pathlib
+import numpy as np
+import yaml
+from PIL import Image
+
+import openslide
+
+
+class WSIReader:
+    def __init__(
+        self,
+        input_dir=pathlib.Path.cwd(),
+        file_name=None,
+        output_dir=pathlib.Path(pathlib.Path.cwd(), "output"),
+        tile_objective_value=20,
+        tile_read_size_w=5000,
+        tile_read_size_h=5000,
+    ):
+        """
+        WSI Reader class to read WSI images
+        Args:
+            input_dir: input path to WSI directory
+            file_name: file name of the WSI
+            output_dir: output directory to save the output, default=os.getcwd()/output
+            tile_objective_value: objective value at which tile is generated, default=20
+            tile_read_size_w: tile width, default=5000
+            tile_read_size_h: tile height, default=5000
+        """
+
+        self.input_dir = pathlib.Path(input_dir)
+        self.file_name = pathlib.Path(file_name).name
+        if output_dir is not None:
+            self.output_dir = pathlib.Path(output_dir, self.file_name)
+
+        self.openslide_obj = openslide.OpenSlide(
+            filename=str(pathlib.Path(self.input_dir, self.file_name))
+        )
+        self.tile_objective_value = np.int(tile_objective_value)
+        self.tile_read_size = np.array([tile_read_size_w, tile_read_size_h])
+        self.objective_power = np.int(
+            self.openslide_obj.properties[openslide.PROPERTY_NAME_OBJECTIVE_POWER]
```

Comment:
not all WSIs may have objective power info. What happens then? We can use a certain pixel resolution from a scanner? If this is done, we should log a warning.

Reply:
That's the main reason I did not include tif for now. I think one way is to input objective power for example for tif files.